AllReadyApp/Web-App/AllReady/Areas/Admin/Features/CampaignManagerInvites/CampaignManagerInviteDetailQueryHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿using AllReady.Areas.Admin.ViewModels.ManagerInvite;
+using AllReady.Models;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+
+namespace AllReady.Areas.Admin.Features.CampaignManagerInvites
+{
+    public class CampaignManagerInviteDetailQueryHandler : IAsyncRequestHandler<CampaignManagerInviteDetailQuery, CampaignManagerInviteDetailsViewModel>
+    {
+        private AllReadyContext _context;
+
+        public CampaignManagerInviteDetailQueryHandler(AllReadyContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<CampaignManagerInviteDetailsViewModel> Handle(CampaignManagerInviteDetailQuery message)
+        {
+            var invite = await _context.CampaignManagerInvites.AsNoTracking()
+                .Include(i => i.Campaign)
+                .Include(i => i.SenderUser)
+                .SingleOrDefaultAsync(i => i.Id == message.CampaignManagerInviteId);
+
+            if (invite == null) return null;
```

Comment:
I'm not too sure why invite here would ever be null. Our code provides the value for `message.CampaignManagerInviteId` via the inviteId  passed to CampaignManagerInviteController.Details action method. The action method is called via our razor views, which themselves are populated from our ViewModels/Query handlers/database...